stdlib/public/runtime/SwiftObject.mm

```diff
@@ -209,6 +209,9 @@ + (void)initialize {
     assert(&objc_debug_isa_class_mask);
     assert(objc_debug_isa_class_mask == SWIFT_ISA_MASK);
   }
+#endif
+  //Class doesn't override -_cfTypeID, so this is the same as NSObject's
+  assert((uintptr_t)(id)[self _cfTypeID] == 1);
```

Comment:
This isn't actually checking anything since `_cfTypeID` is implemented below. You'd want to compare against NSObject's `_cfTypeID` instead. (Maybe in the Foundation overlay?)

Reply:
self here is the Class, not an instance

Reply:
Okay, but you're just testing that root classes are instances of themself. It's not NSObject's implementation.

Reply:
This prints 1:

  Method a = class_getClassMethod([NSString class], @selector(_cfTypeID));
  Method b = class_getInstanceMethod([NSObject class], @selector(_cfTypeID));
  printf("%d\n", a == b);

Instance methods on NSObject are callable as class methods, essentially. Overriding the instance method on a subclass only overrides it for instances of that class, not instances of Class.

Reply:
Instance methods *on a root object* are callable as class methods. A root object that is not NSObject will not get NSObject's instance methods as class methods.

Reply:
Citation: https://github.com/apple/swift-clang/blob/stable/lib/CodeGen/CGObjCMac.cpp#L3614-L3617